### PR TITLE
think this works

### PR DESCRIPTION
**Pull request by Vuwij**
Will prevent rosmon from hanging nodes

### CODE REVIEW

File: src/monitor/monitor.cpp [hunk 1]
```diff
@@ -183,41 +183,45 @@ void Monitor::updateStats()
...
-		node->addCPUTime(stat.utime - oldStat.utime, stat.stime - oldStat.stime);
-		node->addMemory(stat.mem_rss);
+			node->addCPUTime(stat.utime - oldStat.utime, stat.stime - oldStat.stime);
+			node->addMemory(stat.mem_rss);
 
-		infoIt->second.active = true;
-		infoIt->second.stat = stat;
+			infoIt->second.active = true;
+			infoIt->second.stat = stat;
+		} catch (const boost::filesystem::filesystem_error& ex) {
```
Comment (gauthamm) [inline src/monitor/monitor.cpp]: why wrap everything on try? Just wrap only the call that can throw the error and continue.

File: src/monitor/monitor.cpp [hunk 2]
```diff
@@ -183,9 +183,16 @@ void Monitor::updateStats()
 
 	for(; it != end; ++it)
 	{
-		fs::path statPath = (*it) / "stat";
-		if(!fs::exists(statPath))
+        fs::path statPath;
```
Comment (gauthamm) [inline src/monitor/monitor.cpp:183]: formatting

File: src/monitor/monitor.cpp [hunk 3]
```diff
@@ -183,9 +183,16 @@ void Monitor::updateStats()
...
+		try {
+			statPath = (*it) / "stat";
+			if(!fs::exists(statPath))
+				continue;
```
Comment (gauthamm) [inline src/monitor/monitor.cpp]: wrap in braces

**Review by alankritapathak — APPROVED** [review]

**Review by gauthamm — APPROVED** [review]

**Review by abolinsky — APPROVED** [review]

File: src/monitor/monitor.cpp (hunk 3 above)
Comment (abolinsky) [inline src/monitor/monitor.cpp]: please always use braces to mark scope.

**Review by abolinsky — CHANGES_REQUESTED** [review]

**Review by abolinsky — APPROVED** [review]

**Pull request merged**